### DISCUSSION TITLE
Bug with fully qualified names of inner class declarations

### DISCUSSION THREAD

**Discussion by JordanLongstaff** [LemonAppDev/konsist#1169] [Report a Problem] [answered]
### Tell us what version of Konsist you use.

0.15.1

### Was this something which used to work for you, and then stopped?

I am trying to get this working for the first time

### Describe the problem

I'm trying to check that a class inherits from a superclass that was declared inside another class, but it's not working. Here's an example.

In some file:
```
package package.one

class A {
    abstract class B {
        // ...
    }
    // ...
}
```

Then, another file:
```
package package.two

import package.one.A

class C : A.B() {
    // ...
}
```

The declaration of `C` should allow `hasParentOf<A.B>()` to return true, but it doesn't. The issue is with the fully qualified name of `A.B`.

### Actual behaviour

The fully qualified name is `A.B`.

### Expected behaviour

The fully qualified name should be `package.one.A.B`.

**Comment by JordanLongstaff** [comment]
In fact, it looks like Konsist generally struggles to recognize references to inner classes defined in imported outer classes. So `C.hasParentOf<A>()` will also return false.

**Comment by nataliapeterwas** [comment, marked as answer]
Thank you for reporting the problem.

We will fix this in the next release :)

https://github.com/LemonAppDev/konsist/pull/1241

**Comment by nataliapeterwas** [comment]
Konsist `0.16.0` has been released! This release contains a fix for this issue :)
